File: files/2025-05-25/CSEC 25052025.R
library(tidyverse)
library(readxl)

path = "files/2025-05-25/Challenge25.xlsx"
input = read_excel(path, range = "B3:B12")
test = read_excel(path, range = "D3:D6")

letters2 = rep(letters, 2) %>% paste0(collapse = "")
sequences = map(1:26, ~ substr(letters2, .x, .x + 3)) %>% unlist()

result = input %>%
  mutate(
    contains_sequence = map_lgl(Words, ~ any(str_detect(.x, sequences)))
  ) %>%
  filter(contains_sequence) %>%
  select(Words)

all.equal(result$Words, test$Words)
# [1] TRUE

File: files/2025-05-25/CSEC 25052025.py
import pandas as pd
import re
import string

path = "files/2025-05-25/Challenge25.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=10)
test = pd.read_excel(path, usecols="D", skiprows=2, nrows=3).rename(columns=lambda col: col.replace('.1', ''))

seqs = [string.ascii_lowercase * 2][0]
seqs = [seqs[i:i+4] for i in range(26)]

def has_seq(word):
    return any(re.search(seq, word) for seq in seqs)

result = input[input['Words'].apply(has_seq)][['Words']].reset_index(drop=True)

print(result.equals(test))